### PR TITLE
More ThreadContext related cleanup

### PR DESCRIPTION
**Pull request by JeremyKuhne**
A little follow-up cleanup for ThreadContext. Visibility cleanup and other minor tweaks.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/10894)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Application.MarshallingControl.cs [hunk 1]
```diff
@@ -5,41 +5,43 @@ namespace System.Windows.Forms;
 
 public sealed partial class Application
 {
-    /// <summary>
-    ///  This class allows us to handle sends/posts in our winformssynchcontext on the correct thread via
-    ///  control.invoke().
-    /// </summary>
-    internal sealed class MarshalingControl : Control
+    internal abstract unsafe partial class ThreadContext
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Application.MarshallingControl.cs:8]: Should the file name be changed to match this being inside ThreadContext now?

File: src/System.Windows.Forms/src/System/Windows/Forms/Application.ThreadWindows.cs [hunk 2]
```diff
@@ -12,50 +12,29 @@ public sealed partial class Application
...
             // We only do visible and enabled windows.  Also, we only do top level windows.
             // Finally, we only include windows that are DNA windows, since other MSO components
             // will be responsible for disabling their own windows.
-            if (PInvoke.IsWindowVisible(hWnd) && PInvoke.IsWindowEnabled(hWnd))
+            if (PInvoke.IsWindowVisible(hwnd) && PInvoke.IsWindowEnabled(hwnd))
             {
-                bool add = true;
-
-                if (_onlyWinForms)
+                if (!_onlyWinForms || Control.FromHandle(hwnd) is not null)
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Application.ThreadWindows.cs:35]: Consider combining this

File: src/System.Windows.Forms/src/System/Windows/Forms/Application.MarshallingControl.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/Application.MarshallingControl.cs:8]: Yeah, but I want the history to stick so I'm going to chase with another PR.

File: src/System.Windows.Forms/src/System/Windows/Forms/Application.ThreadWindows.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/Application.ThreadWindows.cs:35]: I deliberately left this one as I felt it was a bit clearer.

**Review by lonitra — APPROVED** [review]

**Pull request merged**